### ISSUE TITLE
Optional file extension in route without using regex?

### ISSUE THREAD

**Issue by garetmckinley** [fastify/help#26]
Not sure if this is possible, but I figured I'd ask in case I'm missing something obvious. I'm migrating from express, and previously I had a route that looked like this: `/:name.:ext?`. That allowed me to get two params, the `name` and `ext`. Is it possible to define a route like that in fastify?

I don't even necessarily need the extension as an accessible param, but I can't for the life of me figure out how to allow extensions in a route without obstructing the param.

Even when I tried regex, like `/:name(.svg)`, the `name` ends up being `filename.svg`, instead of `filename`.

Apologies if I'm missing something obvious.

**Comment by delvedor** [comment]
Hi! Thank you for trying Fastify!
Unfortunately, I don't understand what are you trying to achieve, can you rephrase, maybe with an example?

If you just need a dynamic route and get the extension you can do something like this:
```js
fastify.get('/static/:file', (req, reply) => {
  const [name, ext] = req.params.file.split('.')
  ...
})
```

**Comment by garetmckinley** [comment]
Alright, so on express I had a route that looked like this:

```js
router.get("/:name.:ext?", async (req, res) => {
  // ...
}
```

Which would grab both the name and extension (if set) as separate params. Currently, yes, I'm handling it exactly like you mentioned, by splitting it. I was just curious if there was a way to handle it without adding that extra step, like in express. Because if there's no extension added to the url, there's no need to run that split operation at all.

**Comment by delvedor** [comment]
I'm not a regex god, unfortunately, so I'm not able to help you a lot in this regard :P
Anyhow, I would suggest avoiding regex in URL since the performances are terrible, `split` should be way faster.
Furthermore, if you don't have a file extension, in the above example `ext` will be `undefined`, as well as the regex parameter.

**Comment by garetmckinley** [comment]
Alright, no problem. This wasn't really a big issue anyhow, I just figured I'd ask since I'm porting a fairly large high performance  api to fastify and wanted to have the code as 1:1 as possible with the express code.

Appreciate the responses :)
